scalableminds/webknossos
# Implement email verification

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz (Does not make a lot of sense because you need to read the mails for testing)

### Steps to test:
- Sign up at webknossos.
- Notice that your mail is not yet verified (you can check this in the database)
  - If the config requires mail verification and the grace period is 0, you will notice that you can not sign in (even after the account was activated)
- Click the link in the mail (log mails to stdout)
- Your mail will be verified (you can check in db multiUser table)
- You can now log in.
- Clicking the link again results in an error.

### TODO

- [ ] Questions about the Verification flow
	- [ ] When do users get the verification mail? (Maybe only after activation by admin)
	- [ ] Add verification link to welcome mail?
	- [ ] When verification is required (settings), after the grace period, users are not allowed to sign in. Don't know if that makes sense (also resending verification link currently needs sign in, at least that should be possible)
- [x] ~~Once flow is clear, also write about it in docs~~
- [x] Write verification mail
- [x] Mark verification keys as used
- [x] What would a good default application.conf look like?
- [x] Update application.conf (Do this immediately before merging)
- [x] Refresh snapshots (Does not work for me with `yarn refresh-all-snapshots`)

### Issues:
- fixes #7040

------
(Please delete unneeded items, merge only when none are left open)
